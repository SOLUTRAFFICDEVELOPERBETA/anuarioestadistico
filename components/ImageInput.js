/* eslint-disable indent */
import React from 'react'
import PropTypes from 'prop-types';
import { CloudUploadTwoTone } from '@material-ui/icons';
import { Grid, makeStyles } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  image: {
    width: '100%',
    maxHeight: '100%',
    objectFit: 'cover'
  },
  imageContainer: {
    maxWidth: '100%',
    padding: '0.5rem',
    border: '3px solid ',
    borderColor: theme.palette.primary.light,
    borderRadius: '10px',
    color: theme.palette.primary.light
  }
}))

/**
 * @description Componente que permite la subida de una imagen
 * @param {any} value
 * @param {func: () => void } onChange
 * Consulte los elementos utilizados en {@link https://material-ui.com/ | Material-ui}
 */
const ImageInput = ({ label, value, onChange }) => {
  const classes = useStyles()
  /**
   * MÃ©todo para recibir y enviar los cambios en la imagen
   * @param {FileList} files Archivos a subir
   */
  const handleChange = files => {
    const [file] = files
    onChange(file)
  }

  return (
    <div >
      <label htmlFor="annex">
        <Grid container alignContent="center" alignItems="center" justify="center">
          <Grid item container justify="center" direction="column" alignContent="center" alignItems="center" xl={12} lg={12} md={12} sm={12} xs={12} className={classes.imageContainer}>
            {value ? (

              <img className={classes.image}
                src={value} />

            ) : (
                <React.Fragment>
                  <CloudUploadTwoTone fontSize="large" color="primary" />
                  <p>{label}</p>
                </React.Fragment>
              )}
          </Grid>
        </Grid>
      </label>
      <input
        id="annex"
        type="file"
        name="annex"
        multiple={false}
        style={{ display: 'none' }}
        onChange={(e) => handleChange(e.target.files)}
      />
    </div>
  )
}

ImageInput.propTypes = {
  value: PropTypes.string,
  label: PropTypes.string,
  onChange: PropTypes.func
}

ImageInput.defaultProps = {
  label: 'Subir File',
  value: '',
  onChange: (file) => console.log(file)
}

export default ImageInput
