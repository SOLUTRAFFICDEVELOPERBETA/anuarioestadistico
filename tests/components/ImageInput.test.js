import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import ImageInput from '../../components/ImageInput';

describe('Pruebas en <ImageInput/>', () => {
    const onChange = jest.fn();
    const wrapper = shallow(

        <ImageInput
            label="Prueba subir img"
            value="https://increasify.com.au/wp-content/uploads/2016/08/default-image.png"
            onChange={onChange}
        />

    )

    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe mostrar la imagen enviada', () => {
        expect(wrapper.find('img').prop('src')).toBe('https://increasify.com.au/wp-content/uploads/2016/08/default-image.png')
    });

    test('debe llamar la función de onChange() con el parámetro enviado ', () => {
        const parametros = {
            target: {
                files: [
                    {
                        file: 'https://www.antiagingya.com/es/wp-content/uploads/2015/01/img-default-autores.jpg',
                        name: 'imgDefault',
                        size: 343565667,
                        type: 'img',
                    }
                ]
            }
        }

        expect(wrapper.find('input').exists()).toBe(true);
        wrapper.find('input').simulate('change', parametros);

        expect(onChange).toHaveBeenCalledTimes(1);
        expect(onChange).toHaveBeenCalledWith(expect.any(Object));
        expect(onChange).toHaveBeenCalledWith(parametros.target.files[0]);
    })
})