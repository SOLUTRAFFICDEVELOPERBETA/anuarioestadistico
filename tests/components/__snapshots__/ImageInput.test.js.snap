// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ImageInput/> Debe mostrarse correctamente 1`] = `
<div>
  <label
    htmlFor="annex"
  >
    <WithStyles(ForwardRef(Grid))
      alignContent="center"
      alignItems="center"
      container={true}
      justify="center"
    >
      <WithStyles(ForwardRef(Grid))
        alignContent="center"
        alignItems="center"
        className="makeStyles-imageContainer-2"
        container={true}
        direction="column"
        item={true}
        justify="center"
        lg={12}
        md={12}
        sm={12}
        xl={12}
        xs={12}
      >
        <img
          className="makeStyles-image-1"
          src="https://increasify.com.au/wp-content/uploads/2016/08/default-image.png"
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </label>
  <input
    id="annex"
    multiple={false}
    name="annex"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
</div>
`;
